src/EFCore.MySql/Infrastructure/CharSetConfig.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Pomelo Foundation. All rights reserved.
+// Licensed under the MIT. See LICENSE in the project root for license information.
+
+using Pomelo.EntityFrameworkCore.MySql.Storage.Internal;
+
+namespace Pomelo.EntityFrameworkCore.MySql.Infrastructure
+{
+    public class CharSetConfig
+    {
+        public CharSetInfo ServerDefaultCharSetInfo { get; private set; }
+
+        public string ServerDefaultCharSet
+        {
+            get => ServerDefaultCharSetInfo?.CharSetName;
+            set => ServerDefaultCharSetInfo = new CharSetInfo(value);
+        }
+
+        public CharSetInfo AnsiCharSetInfo { get; private set; }
+
+        public string AnsiCharSet
+        {
+            get => AnsiCharSetInfo?.CharSetName;
+            set => AnsiCharSetInfo = new CharSetInfo(value);
+        }
+
+        public CharSetInfo UnicodeCharSetInfo { get; private set; }
```

Comment:
I think it would be better to have the charsets set to some value by default as it would make the model independent from the server. One less thing for users to worry about unless they opt-in to follow the server default.